### PR TITLE
Add e2e test to test Integration System deletion

### PR DESCRIPTION
**Pull request by KarolJaksik**
**Description**

Changes proposed in this pull request:

- Add custom ConstraintViolation error
- Make sure that if IntegrationSystem deletion fails SystemAuths are not invalidated in hydra
- ...

**Related issue(s)**
See #442

### CODE REVIEW

**Comment by dbadura** [comment]
Did you created issue about concept of background job, which invalidates marked credentials as invalid in hydra?

**Comment by KarolJaksik** [comment]
> Did you created issue about concept of background job, which invalidates marked credentials as invalid in hydra?

Yes, you can see it [here](https://github.com/kyma-incubator/compass/issues/479)

File: tests/end-to-end/pkg/gql/graphqlizer.go [hunk 1]
```diff
@@ -48,9 +48,23 @@ func (g *Graphqlizer) ApplicationCreateInputToGQL(in graphql.ApplicationCreateIn
...
 
+func (g *Graphqlizer) ApplicationUpdateInputToGQL(in graphql.ApplicationUpdateInput) (string, error) {
+	return g.genericToGQL(in, `{
+		name: "{{.Name}}",
+		{{- if .Description }}
+		description: "{{.Description}}",
+		{{- end }}
+		{{- if .HealthCheckURL }}
+		healthCheckURL: "{{ .HealthCheckURL }}"
+		{{- end }}
```
Comment (dbadura) [inline tests/end-to-end/pkg/gql/graphqlizer.go]:
```suggestion
		{{- end }}
		{{- if .IntegrationSystemID }}
			integrationSystemID: "{{ .IntegrationSystemID }}",
		{{- end }}

File: components/director/internal/repo/delete.go [hunk 2]
```diff
@@ -78,8 +80,13 @@ func (g *universalDeleter) unsafeDelete(ctx context.Context, tenant *string, con
 	allArgs := getAllArgs(tenant, conditions)
 	res, err := persist.Exec(stmtBuilder.String(), allArgs...)
 	if err != nil {
+		pqerr, ok := err.(*pq.Error)
+		if ok {
+			return specificPqError(pqerr)
```
Comment (crabtree) [inline components/director/internal/repo/delete.go]: Consider handling the error similarly to `os.IsNotExist(err)`

File: components/director/pkg/apperrors/errors.go [hunk 3]
```diff
@@ -74,6 +74,36 @@ func IsNotUnique(err error) bool {
...
+	return &constraintViolationError{
+		table: table,
+	}
+}
+
+func (e *constraintViolationError) Error() string {
+	return fmt.Sprintf("this object still has %s referenced by it", e.table)
+}
+
+func (constraintViolationError) IsConstraintViolation() {}
```
Comment (crabtree) [inline components/director/pkg/apperrors/errors.go]:
```suggestion
func (constraintViolationError) ConstraintViolation() {}
```

File: components/director/pkg/apperrors/errors.go [hunk 4]
```diff
@@ -74,6 +74,36 @@ func IsNotUnique(err error) bool {
 	return false
 }
 
+type ConstraintViolation interface {
+	IsConstraintViolation()
```
Comment (crabtree) [inline components/director/pkg/apperrors/errors.go]:
```suggestion
	ConstraintViolation()
```

File: components/director/internal/persistence/const.go [hunk 5]
```diff
@@ -9,4 +9,6 @@ const (
 	PersistenceCtxKey persistenceCtxKey = "PersistenceCtx"
 	// UniqueViolation is the error code that happens when the Unique Key is violated
 	UniqueViolation pq.ErrorCode = "23505"
+	//ConstraintViolation is the class of errors that happens when any constraint is violated
+	ConstraintViolation = "23"
```
Comment (crabtree) [inline components/director/internal/persistence/const.go]:
```suggestion
	ConstraintViolation pq.ErrorCode = "23"
```

File: components/director/internal/persistence/errors.go [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+import (
+	"github.com/lib/pq"
+)
+
+func IsConstraintViolation(err error) bool {
+	if err == nil {
+		return false
+	}
+
+	pqerr, ok := err.(*pq.Error)
```
Comment (crabtree) [inline components/director/internal/persistence/errors.go]: How about this?
```go
	if pqerr, ok := err.(*pq.Error); ok {
		return pqerr.Code.Class() == ConstraintViolation
	}
```

**Pull request merged**